javascript/packages/fury/lib/platformBuffer.ts

```diff
@@ -17,23 +17,38 @@
  * under the License.
  */
 
-import { isNodeEnv } from "./util";
+import { hasBuffer } from "./util";
+
+export type SupportedEncodings = "latin1" | "utf8";
 
 export interface PlatformBuffer extends Uint8Array {
-  latin1Slice(start: number, end: number): string;
-  utf8Slice(start: number, end: number): string;
-  latin1Write(v: string, offset: number): void;
-  utf8Write(v: string, offset: number): void;
+  toString(encoding?: SupportedEncodings, start?: number, end?: number): string;
+  write(string: string, offset: number, encoding?: SupportedEncodings): void;
   copy(target: Uint8Array, targetStart?: number, sourceStart?: number, sourceEnd?: number): void;
 }
 
 export class BrowserBuffer extends Uint8Array implements PlatformBuffer {
+  write(string: string, offset: number, encoding: SupportedEncodings = "utf8"): void {
+    if (encoding === "latin1") {
+      return this.latin1Write(string, offset);
+    }
+    return this.utf8Write(string, offset);
+  }
+
+  toString(encoding: SupportedEncodings = "utf8", start = 0, end = this.length): string {
+    if (encoding === "latin1") {
+      return this.latin1Slice(start, end);
+    }
+    return this.utf8Slice(start, end);
+  }
+
   static alloc(size: number) {
     return new BrowserBuffer(new Uint8Array(size));
   }
 
   latin1Write(string: string, offset: number) {
-    for (let index = 0; index < string.length; index++) {
+    let index = 0;
```

Comment:
here I originally want return the index, because the `Buffer.prototype.write` will return the writes counts

Reply:
but i found it not necessary to do this, so i remove the return statement